geopandas/io/sql.py

```diff
@@ -96,3 +96,222 @@ def load_geom_text(x):
+    import io
+    import csv
+
+    # Convert columns to lists and make a generator
+    args = [list(gdf[i]) for i in gdf.columns]
+    if index:
+        args.insert(0, list(gdf.index))
+
+    data_iter = zip(*args)
+
+    # get list of columns using pandas
+    keys = tbl.insert_data()[0]
+    columns = ", ".join('"{}"'.format(k) for k in list(keys))
+
+    s_buf = io.StringIO()
+    writer = csv.writer(s_buf)
+    writer.writerows(data_iter)
+    s_buf.seek(0)
+
+    conn = engine.raw_connection()
+    cur = conn.cursor()
+
+    try:
+        # If appending to an existing table, temporarily change
+        # the srid to 0, and update the SRID afterwards
+        if if_exists == "append":
+            sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{crs})".format(
+                schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, crs=0
+            )
+            cur.execute(sql)
+
+        sql = "COPY {} ({}) FROM STDIN WITH CSV".format(tbl.table.fullname, columns)
+        cur.copy_expert(sql=sql, file=s_buf)
+
+        # SRID needs to be updated afterwards as the current approach does not support
+        # the use of EWKT geometries.
+        sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{srid})".format(
+            schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, srid=srid
+        )
+        cur.execute(sql)
+        conn.commit()
+
+    except Exception as e:
+        conn.connection.rollback()
+        raise e
+    conn.close()
+
+
+def write_postgis(
+    gdf, con, table, if_exists="fail", schema=None, dtype=None, index=False
+):
+    """
+    Upload GeoDataFrame into PostGIS database.
+
+    Parameters
+    ----------
+
+    gdf : GeoDataFrame
+        GeoDataFrame containing the data for upload.
+    con : sqlalchemy.engine.Engine.
+        Connection engine to the database.
+    if_exists : str
+        What to do if table exists already: 'replace' | 'append' | 'fail'.
+    schema : db-schema
+        Database schema where the data will be uploaded (default: 'public').
+    dtype : dict of column name to SQL type, default None
+        Optional specifying the datatype for columns. The SQL type should be a
+        SQLAlchemy type, or a string for sqlite3 fallback connection.
+    index : bool
+        Store DataFrame index to the database as well.
+    """
+    from geoalchemy2 import Geometry
+    from shapely.geometry import MultiLineString, MultiPoint, MultiPolygon
+
+    gdf = gdf.copy()
+    geom_name = gdf.geometry.name
+    if schema is not None:
+        schema_name = schema
+    else:
+        schema_name = "public"
+
+    # Get srid
+    srid = get_srid_from_crs(gdf)
+
+    # Check geometry types
+    geometry_type, contains_multi_geoms = get_geometry_type(gdf)
+
+    # Build target geometry type
+    if contains_multi_geoms:
+        target_geom_type = "Multi{geom_type}".format(geom_type=geometry_type)
```

Comment:
Now I see that by _Geometry Collection_ above you meant Multi- geometry only. So GeometryCollections has to be somehow taken care of and I would prefer not using "collection" for Multi- as it is confusing :).

Reply:
This is fixed in the latest version.